wordpress-mobile/WordPress-iOS
# Do not make another attempt if the HTTP request was cancelled

Fixes #20379, which is a crash that may happen during log out. See [this comment](https://github.com/wordpress-mobile/WordPress-iOS/issues/20379#issuecomment-1481995663) for the potential cause of this crash.

This crash _should_ be reproducible in the app by simply tapping the log out button, if you set up a proxy and delay `/me/settings` API call's response for a few seconds. I tried to do that using mitmproxy but didn't find a "delay response" feature in it and gave up.

## Test instructions

https://github.com/wordpress-mobile/WordPress-iOS/commit/f280227989f1e429bb06d985cd858c8ff3e7db5c reproduces this crash in a unit test. Check out this commit locally and run the added unit test to verify the app indeed crashes.

If the CI passes then we should be all good. Or, you can checkout the latest commit in this PR branch and run the same test again to verify the app no longer crashes.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
